tests/cpp/test_mma.cpp

```diff
@@ -496,6 +496,157 @@ TEST_P(HopperRS, SingleTileWithTMALoadStore) {
   EXPECT_TRUE(at::allclose(cg_outputs[0], tref, 1e-5, 1e-5));
 }
 
+using HopperMmaRSStMatrixTestParams = std::tuple<
+    MmaMacro,
+    PrimDataType,
+    MmaLayout,
+    MmaInputSmemSwizzle,
+    std::vector<int>>;
+
+class HopperRSStmatrix
+    : public HopperBase,
+      public ::testing::WithParamInterface<HopperMmaRSStMatrixTestParams> {
+ protected:
+  MmaLayout layout;
+  MmaMacro macro;
+  PrimDataType dtype;
+  MmaInputSmemSwizzle swizzle_b;
+  std::vector<int> tile_sizes;
+
+  void SetUp() override {
+    HopperBase::SetUp();
+    macro = std::get<0>(GetParam());
+    dtype = std::get<1>(GetParam());
+    layout = std::get<2>(GetParam());
+    swizzle_b = std::get<3>(GetParam());
+    tile_sizes = std::get<4>(GetParam());
+  }
+};
+
+TEST_P(HopperRSStmatrix, SingleTileWithTMALoadStoreStMatrix) {
```

Comment:
Should we remove `SingleTileWithTMALoadStore` in favor of this? There are too many hopper mma tests, and we should try to remove some without hurting coverage too much.